app/code/Magento/Catalog/view/adminhtml/web/js/components/dynamic-rows-tier-price.js

```diff
@@ -28,6 +28,33 @@ define([
             });
 
             this.labels(labels);
+        },
+
+        /**
+         * Change page
+         *
+         * @param {Number} page - current page
+         */
+        changePage: function (page) {
+            this.clear();
+
+            if (page === 1 && !this.recordData().length) {
```

Comment:
Hi @saravananvelu. Thanks for the collaboration.
This issue reproduced in all places where dynamic-rows component is used, so will be great if you fix this issue in the more general way, in the parent class. 

Also, you can call the method from the parent class (`this._super()`) instead of copy-paste it

Reply:
I have updated the code in parent class and removed the same from class